### PR TITLE
Structural PR

### PR DESCRIPTION
**Pull request by arthurpaulino**
* Add remappings for easier imports
* Fix references
* Clean up copied files

### CODE REVIEW

**Review by storojs72 — CHANGES_REQUESTED** [review]
As far as I see, two things are introduced essentially:
1) Usage of `pasta-solidity` as a third-party dependency;
2) Removing `PallasContract` wrapper and using Pallas / Vesta libraries directly. 

Regarding first point - my initial motivation behind including Pallas / Vesta sources is following: it essentially allows better controlling the pasta building block, which is an important cryptographic component. By using third-party dependency, you usually treat it as a black-box and if you need some changes there it is difficult. Obviously, there's a tradeoff - as soon as we should pay some cost for maintaining additional code. Summarising this, we can switch to third-party dependency later, once we successfully integrate it into future implementation of Nova verifier, and we will know that no changes are required on pasta curves code by Nova.

Regarding Pallas / Vesta libraries direct usage in tests - I think we will still need some contracts / scripts when we come to interacting with the actual network, so this mock can be used as a basis actually. 

So, I don't mind refactoring tests to use libraries directly and I would still prefer keeping Pallas / Vesta libraries sources at least until we are convinced that no changes are required by Nova verifier

File: test/pasta-curves-tests.t.sol [hunk 1]
```diff
@@ -185,7 +176,7 @@ contract PastaCurvesContractTests is Test {
 
         uint256 scalar = 0x0716ea7ecbe554f483e62dc4a31ef8d080b35ea748906aa770491c144257819a;
 
-        Pallas.PallasProjectivePoint memory p1MulScalar = pallas.projectiveScalarMul(p1, scalar);
+        Pallas.PallasProjectivePoint memory p1MulScalar = Pallas.projectiveScalarMul(p1, scalar);
```
Comment (storojs72) [inline test/pasta-curves-tests.t.sol:179]: I guess, here it should be `Pallas.scalarMul`, as soon as we call library's function directly

**Review by storojs72 — APPROVED** [review]
LGTM

File: test/pasta-curves-tests.t.sol (hunk 1 above)
Comment (arthurpaulino) [reply test/pasta-curves-tests.t.sol:179]: Yes, but that code is commented out. The line change was due to a "replace all" command

**Pull request merged**